README.md

```diff
@@ -186,6 +187,31 @@ method takes one argument:
 
 2. `map` must return a value of the same Functor
 
+### Contravariant Functor
+
+1. `u.contramap(a => a)` is equivalent to `u` (identity)
+2. `u.contramap(x => f(g(x)))` is equivalent to `u.contramap(f).contramap(g)`
+(composition)
+
+#### `contramap` method
+
+```hs
+contramap :: Contravariant f => f a ~> (b -> a) -> f b
```

Comment:
Are there any examples of contra variance on value based ADTs?

I have only seen them on function based bits that have their returns fixed. I know it moves away from the Haskell definition, but would it make sense to adopt a signature more like:

`Contravariant f => f a b ~> (c -> a) -> f c b`

or

`Contravariant f => f (a -> b) ~> (c -> a) -> f (c -> b)`

Or is that to limiting for the spec?